functionary/prompt_template/base_template.py

```diff
@@ -228,6 +228,10 @@ def get_force_function_call_prefix(self, function_name: str):
         """
         raise NotImplementedError
 
+    def get_tool_choice_required_prefix(self):
```

Comment:
So we also need to use this function in `def parse_assistant_response` and also `update_response_state_from_delta_text`